### PR TITLE
feat: update readme

### PR DESCRIPTION
**Pull request by dreinhardt-terminus**
- [x] I promise to make sure that this PR is correct before merging it.

#### Does this change have potential to cause downtime?

- [x] No - This is a low risk change
- [ ] Yes - This is high risk change and must be elevated or higher and you need more approvals

### CODE REVIEW

**Review by cduddikunta — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,4 +1,501 @@
...
+steps:
+  - name: Get and Set Secrets, Env Variables, and TF_VARs
+    uses: GetTerminus/terminus-github-actions/internal-actions/aws-get-secrets@v2
+    with:
+      aws_account_id: ${{ inputs.aws_account_id }}
+      aws_role_name: "infra-eks"
+      aws_role_path: "/terraform"
+      secrets: |
+        DATADOG_API_KEY, datadog-api-key
+        DATADOG_APP_KEY, datadog-app-key
```
Comment (billjamison) [inline README.md:39]: Which is the secret id / name, and which is the env var name? The text suggests the structure is `<secret, env-var>[,<secret, env-var]`, but the example -appears- to be the other way round.

File: README.md [hunk 2]
```diff
@@ -1,4 +1,501 @@
...
+
+    # below is an example of how you would use the headers
+  - name: GitHub API Example
+    run: |
+      curl -s -L \
+        -H "${{ env.GH_ACCEPT }}" \
+        -H "${{ env.GH_AUTH }}" \
+        -H "${{ env.GH_API }}" \
+        https://api.github.com/repos/GetTerminus/example
+```
```
Comment (billjamison) [inline README.md:67]: This shows how to consume them, but not how to set them?

File: README.md [hunk 3]
```diff
@@ -1,4 +1,501 @@
...
+    uses: GetTerminus/terminus-github-actions/internal-actions/github-artifacts-download@v2
+    with:
+      artifact_name: 'tf-provider-cache-key-file'
+      github_repo: ${{ github.repository }}
+      github_current_branch: ${{ env.CURRENT_BRANCH }}
+      github_default_branch: ${{ inputs.event.repository.default_branch }}
+```
+
+### GitHub - Artifacts Enable Upload
+Determines whether artifacts should be uploaded, used for terraform caching.
```
Comment (billjamison) [inline README.md:103]: Typo: ...uploaded**. U**sed for...

File: internal-actions/github-artifacts-find/action.yaml [hunk 4]
```diff
@@ -1,12 +1,12 @@
 name: 'Github Find Artifact'
-description: 'Finds an artifact using repo, current and default branch, and artifact name.'
+description: 'Finds artifacts from a GitHub repo, tries the current branch then the default branch if an artifact is not found.'
 inputs:
   #Required
   artifact_name:
-    description: '[string] (i.e. tf-cache-key-files) Name of the artifact.'
+    description: '[string] (i.e. artifact-name) Name of the artifact.'
     required: true
```
Comment (billjamison) [inline internal-actions/github-artifacts-find/action.yaml:7]: I agree there is word-smithing that needs to be done here. However, it looks like concrete examples have been replaced with abstract ones. I wonder if that's the best choice? 

For example, replacing the concrete artifact name 'tf-cache-key-files' with the abstract identifier 'artifact-name'. These abstract examples seem particularly ... interesting ... given that they are now re-statements of the rest of the description (e.g., `'... (i.e., GetTerminus/**repo-name**) **Name of the repo** where ...'`

**Pull request merged**